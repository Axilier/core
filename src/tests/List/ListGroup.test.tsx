import React from 'react';
import { mount } from 'enzyme';
import { ListGroup, ListItem } from '../../index';
import { expectToThrow } from '../expectToThrow';

const onClick = jest.fn();
const listGroup = mount(
    <ListGroup
        listItem={
            <ListItem key={'item1'} onClick={onClick}>
                <></>
            </ListItem>
        }
    >
        <ListItem key={'item2'}>
            <></>
        </ListItem>
    </ListGroup>,
);
const cleanUp = () => listGroup.setProps({ dropdown: true });

describe('when I render the List in an application', () => {
    it('should render correctly', () => {
        expect(listGroup).toMatchSnapshot();
    });
    describe('when I provide `listItem` of a type not ListItem', () => {
        it('should throw an error', () => {
            expectToThrow(
                () =>
                    mount(
                        <ListGroup listItem={<div />}>
                            <ListItem>
                                <></>
                            </ListItem>
                        </ListGroup>,
                    ),
                'The listItem prop must be of type listItem',
            );
        });
    });
    describe('when I provide `children` of a type not ListItem', () => {
        it('should throw an error', () => {
            expectToThrow(
                () =>
                    mount(
                        <ListGroup
                            listItem={
                                <ListItem>
                                    <></>
                                </ListItem>
                            }
                        >
                            <div />
                            <div />
                        </ListGroup>,
                    ),
                'Children must be of type ListItem',
            );
        });
    });
    describe('when the groups `listItem` is clicked', () => {
        describe('when the `onClick` for the `listItem` is defined', () => {
            beforeAll(() => listGroup.childAt(0).simulate('click'));
            it('should open the listItems', () => {
                expect(listGroup.childAt(1).props().style).toHaveProperty(
                    'height',
                    '40px',
                );
            });
            it('should run the provided onClick function', () => {
                expect(onClick).toHaveBeenCalled();
            });
        });
        describe('when the `onClick` for the `listItem` is not defined', () => {
            beforeAll(() => {
                listGroup.setProps({
                    listItem: (
                        <ListItem key={'item1'}>
                            <></>
                        </ListItem>
                    ),
                });
                listGroup.childAt(0).simulate('click');
            });
            it('should do nothing', () => {
                expect(true).toBeTruthy();
            });
        });
    });
});
