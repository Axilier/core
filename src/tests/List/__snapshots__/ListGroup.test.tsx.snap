// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the List in an application should render correctly 1`] = `
<ListGroup
  listItem={
    <ListItem
      onClick={[MockFunction]}
    >
      <React.Fragment />
    </ListItem>
  }
>
  <ListItem
    dropdown={false}
    key="item1"
    onClick={[Function]}
  >
    <Layout
      className="listItem"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      orientation="row"
      style={
        Object {
          "backgroundColor": "#F9F9F9",
          "paddingLeft": "30px",
        }
      }
    >
      <div
        className="listItem"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "backgroundColor": "#F9F9F9",
            "display": "flex",
            "flexDirection": "row",
            "paddingLeft": "30px",
          }
        }
      >
        <div
          className="listItemChildren"
        />
        <div
          className="endFragment"
        >
          <svg
            className="dropdown-test"
            fill="black"
            height="16"
            style={
              Object {
                "transform": "none",
              }
            }
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.204 5h9.592L8 10.481 3.204 5zm-.753.659 4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659z"
            />
          </svg>
        </div>
      </div>
    </Layout>
  </ListItem>
  <div
    className="listGroup"
    style={
      Object {
        "height": "0px",
      }
    }
  >
    <List>
      <Layout
        orientation="column"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "width": "100%",
            }
          }
        >
          <ListItem
            child={true}
            key="item2"
          >
            <Layout
              className="listItem"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              orientation="row"
              style={
                Object {
                  "backgroundColor": "#F9F9F9",
                  "paddingLeft": "40px",
                }
              }
            >
              <div
                className="listItem"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "backgroundColor": "#F9F9F9",
                    "display": "flex",
                    "flexDirection": "row",
                    "paddingLeft": "40px",
                  }
                }
              >
                <div
                  className="listItemChildren"
                />
                <div
                  className="endFragment"
                >
                  <div
                    style={
                      Object {
                        "height": "16px",
                        "width": "16px",
                      }
                    }
                  />
                </div>
              </div>
            </Layout>
          </ListItem>
        </div>
      </Layout>
    </List>
  </div>
</ListGroup>
`;
